### PR TITLE
inject request stack for symfony 6

### PR DESCRIPTION
**Pull request by tuxes3**
To support symfony 6

### CODE REVIEW

File: Core/H5PSymfony.php [hunk 1]
```diff
@@ -214,11 +227,10 @@ public function setInfoMessage($message)
      */
     public function getMessages($type)
     {
-        if (!$this->session->getFlashBag()->has($type)) {
+        if ($this->session === null || !$this->session->getFlashBag()->has($type)) {
```
Comment (jorisdugue) [inline Core/H5PSymfony.php:230]: replace this row is better 😄 

```suggestion
        if (!$this->session || !$this->session->getFlashBag()->has($type)) {
```

**Review by jorisdugue — CHANGES_REQUESTED** [review]
Thanks for making these changes but there are a few changes to make to improve code quality and readability

File: Core/H5PSymfony.php [hunk 2]
```diff
@@ -93,10 +96,16 @@ public function __construct(H5POptions $options,
         $this->editorStorage = $editorStorage;
         $this->tokenStorage = $tokenStorage;
         $this->manager = $manager;
-        $this->session = $session;
         $this->authorizationChecker = $authorizationChecker;
         $this->eventDispatcher = $eventDispatcher;
         $this->router = $router;
+        try {
+            $this->session = $requestStack === null ? null : $requestStack->getSession();
```
[the remark this answers is not in the document]
Comment (tuxes3) [reply Core/H5PSymfony.php]: Hey. I had a case where `$requestStack->getSession();` did throw the `SessionNotFoundException`. Also there is no method to check if the session is present or not. Furthermore we did change the code to support no session, which would be the case here if no session was found!

For these reasons i did not remove the `try ... catch`.

[the remark this answers is not in the document]
Comment (jorisdugue) [reply Core/H5PSymfony.php]: Hey @tuxes3 ,
I kinda suspected that, we have no choice but to silence the exception for once otherwise it won't work

**Review by jorisdugue — APPROVED** [review]
Thanks for the edits, looks good to me :)

**Pull request merged**